### ISSUE TITLE
Why no `MaskedScatterOffset`?

### ISSUE THREAD

**Issue by fbarbari** [google/highway#1753]
Hi all,
maybe I am missing some bit of knowledge, but why are functions like `MaskedScatterOffset`, `ScatterOffsetN`, `MaskedGatherOffset` and `GatherOffsetN` missing?

**Comment by jan-wassenberg** [comment]
We haven't added those because there are now multiple versions of scatter (Masked, soon also Masked..Or) and doubling the number of ops to support byte offsets did not seem useful. Do you rely on byte offsets? Is it perhaps an option to right-shift your offsets to get indices, or would that be on the critical path for your kernels?

**Comment by fbarbari** [comment]
Actually, I don't rely on byte offsets, I was just curious to know since there is only one masked op and three non-masked. I was translating a kernel into Highway in the most literal way and came across this missing function so I tried to understand if it indeed was a missing function or it was a design choice or a way to say "if we don't provide it, you're doing it wrong".
For the same reason, I was wondering why offsets and indices for Gather and Scatter need to be signed integers instead of unsigned.

**Comment by jan-wassenberg** [comment]
OK :) We'll add some documentation to address those questions. It's indeed a design choice.
The reason for signed is to draw attention to the unfortunate design choice of RVV, which differs from the way everyone else interprets indices as signed.

**Comment by fbarbari** [comment]
Thank you very much for the clarification. I also have another question/proposal about other masked operations that highway doesn't have (AFAIK). May I post it here or is it better to open another issue?

**Comment by jan-wassenberg** [comment]
You're welcome, thanks for pointing that out so we can document it.
Copybara is going to auto-close this issue once the change is merged, so better to open another :)

Some general observations:
- we are happy to add ops whenever there is a documented need or benefit.
- for masking, my current thinking is that the main benefit is in avoiding exceptions or perhaps FP assists from denormals. That is why we added first for memory, recently also for arithmetic.
-  Sometimes clang already generates masked instructions for AVX-512 given `IfThenElse(mask, op(), default)`.

Happy to discuss your use case further :)

**Comment by fbarbari** [comment]
>  recently also for arithmetic.
I didn't notice. It was exactly what I was going to ask. Thank you again.
